packages/tasit-act/src/tasitAct.test.js

```diff
@@ -0,0 +1,83 @@
+
+describe.skip("Contract", function() {
+  let simpleStorage;
+
+  beforeEach("should connect to and existing contract", async () => {
+    // Connection with blockchain
+    // Ethers.js allows sets a "connector" (provider or already connected wallet) on constructor of Contract
+    // or later with contract.connect() method.
+    // How connection should be made in here?
+    // - Make sense always uses `tasit-account` to create connected account instead of interact with provider object?
+    // - If Account encapsulates the provider object, Contract will connect with blockchain thru Account.
+    // I think that the point here is about how/if we will expose Provider object
+    simpleStorage = new Contract(contractAddress, contractABI);
+    simpleStorage.connect(account); // or new Contract(addr, abi, account)
+    expect(simpleStorage).to.exist;
+    expect(simpleStorage.address).to.equal(contractAddress);
+  });
+
+  it("should call a read-only contract method", async () => {
+    const value = await simpleStorage.getValue();
+    expect(value).to.exist;
+  });
+
+  it("should call a write contract method (send tx)", async () => {
+    var rand = Math.floor(Math.random() * Math.floor(1000));
+    const tx = await simpleStorage.setValue(rand);
+
+    // How to wait a tx?
+    // Ethers uses:
+    // await provider.waitForTransaction(updateValueTx.hash);
+    // Some possibilities:
+    await simpleStorage.waitForTransaction(tx.hash);
+    // or
+    // await simpleStorage.getAccount().waitForTransaction(tx.hash);
+    // or even (too much chained)
+    // await simpleStorage.getAccount().getProvider().waitForTransaction(tx.hash);
+
+    const value = await simpleStorage.getValue();
+    expect(value).to.equal(rand);
+  });
+
+  it("should send a signed message", async () => {
+    // Steps to do that:
+    // 1. Encode contract function with param(s);
+    // 2. Create rawTx with data above;
+    // 3. Sign rawTx
+    // 4. Broadcast signedTx
+
+    // Step 1
+    // Web3.js 1.0: data = simpleStorage.methods.setValue(rand).encodeABI()
+    // Ether.js:
+    const data = simpleStorage.interface.functions.setValue.encode([rand]); // How we'll do that?
+
+    // Step 2
+    // A basic tx contains these params:
+    // gasLimit: I'm almost sure that the gasLimit can be calculated automatically
+    // to: simpleStorage.address
+    // data: Encoded function call
+    // nonce: Can be get easly from wallet.address
+    //
+    // That said, we can do Step 1 & 2 with one command like that:
+    // In the case, command above will be unecessary
+    const rawTx = simpleStorage.interface.functions.setValue(rand).toRawTx();
+
+    // Step 3
+    const signedTx = simpleStorage.getAccount().sign(rawTx);
+    // or we can do 1..3 steps:
+    // const signedTx = simpleStorage.interface.functions.setValue(rand).toSignedTx();
+
+    // Step 4
+    const sentTx = simpleStorage.sendTransaction(signedTx); // Same as provider.sendTransaction
```

Comment:
Is there a separate function for broadcasting it for meta-transaction usage vs. "sending" it? If so, we want to broadcast it here instead, I think

Reply:
I've found this great answer: https://ethereum.stackexchange.com/questions/62578/meta-transaction-with-web3
Seems that broadcasting should be made to a relay responsible to pay for the transaction gas. 
I need to research more on that.

Reply:
Yep, that's exactly right. So `tasit-act` should have two functions, one that does a regular transaction and one that broadcasts a potential transaction. Either that or it's a single function with a param that affects which happens. I'm leaning towards single function with a "free / not free boolean", since that seems to abstract away blockchain details better.

Reply:
And ethers.js has a separate function for broadcasting, is that right?

Reply:
No, not really. It seems that the `account` should sign the message even if it doesn't pay. The broadcasting is made outside of Ether.js using a regular http request or another way that relay serves.
The relayer will sign too and then send a transaction to the `identity-contract`.

I need to research more on that to give a sure and detailed answer to you.

Reply:
Yeah, for a meta-transaction, the account definitely needs to sign the message even if it doesn't pay.

Okay, we can make our own small js function from scratch for broadcasting that's configurable for the most popular approaches, in that case.

To start we can prioritize regular transactions over meta-transactions as long as we put in some placeholder functions for broadcasting and test the broadcast function for meta-transactions with a mock.

Yeah, the relayer sends a transaction to the identity-contract in the case where it's using identity contracts. You can also use a raw meta-transaction and someone can pay for the transaction on your behalf without an identity contract. I think we'll want to support both options.

Reply:
Related to: https://github.com/tasitlabs/TasitSDK/pull/71#discussion_r243331285

Reply:
extracted to #88

Reply:
Resolving this